### DISCUSSION TITLE
Dota Plus Win Probabilities

### DISCUSSION THREAD

**Discussion by yurtsevm** [ValveSoftware/Dota2-Gameplay#3172] [Ideas]
Hello all,
This is my first time using github and suggesting an idea to Valve/DotA 2, so I really hope that it is the appropriate place to do it. Today while spectating a game, I came to an idea of suggesting a new way of calculating win probabilities in live games. If that is how it is calculated, please kindly tell me. I always thought that an already supervised learning algorithm decides on the probabilities. (Please note that this approach does not necessarily have to see and know the exact same game with the heroes, levels, items etc. to come up with the probabilities.)

My idea is a little more different, here it goes:
At any given point in a game, take a snap-shot of the current game. All the heroes, hero levels, hero items, cooldowns, skills, talents, positions, roshan spawn time, neutral camps, teams' tower situations, outposts etc.
And then feed it to some kind of AI bot environment, idk, OpenAI seems like a good idea since it already has an idea on how to play the game.
Let the bots play some certain amount of games, say 1000, and then report the times Radiant win out of a 1000, and for the Dire side as well. Even with a slight knowledge on AI/ML and software, my guess is that taking the snapshot of the game and feeding it to your environment should not be a problem. Also, running your created "game" 1000 times in a small amount of time shouldn't be an issue too. 
Basically, like a Monte-Carlo simulation with some smart AI bots.

I would love to hear and discuss on this idea of mine with y'all.
Best,
